packages/@aws-cdk/aws-servicecatalog/lib/constraints.ts

```diff
@@ -0,0 +1,159 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as sns from '@aws-cdk/aws-sns';
+import { AcceptLanguage } from './common';
+import { IProduct } from './product';
+
+/**
+ * Properties for governance mechanisms and constraints.
+ */
+export interface ConstraintProps {
+
+  /**
+   * The language code.
+   * @default - No accept language provided
+   */
+  readonly acceptLanguage?: AcceptLanguage;
+
+  /**
+   * The description of the constraint.
+   * @default - No description provided
+   */
+  readonly description?: string;
+
+  /**
+   * A reference to product.
+   */
+  readonly product: IProduct;
+}
+
+/**
+ * Properties for LaunchNotificationConstraint.
+ */
+export interface EventNotificationsProps extends ConstraintProps {
+
+  /**
+   * A list of SNS Topics to notify on Stack Events.
+   */
+  readonly topics: sns.ITopic[];
+}
+
+/**
+ * Properties for LaunchRoleConstraint.
+ */
+export interface LaunchRoleProps extends ConstraintProps {
+
+  /**
+   * A reference to a Role
+   */
+  readonly role: iam.IRole;
+}
+
+/**
+ * Rule for Service Catalog Template Constraint.
+ */
+interface Rule {
+
+  /**
+   * List of assertions
+   */
+  readonly Assertions: Assertion[],
```

Comment:
Wrong casing.

Reply:
We applied a linting rule exception here, otherwise we have to create an extra function that can go in and capitalize the first letter of all the keys in the map.  Would that be okay?

Reply:
I would change to lowercase property names, and write that extra function. Should be an easy refactor with the tests you have already.